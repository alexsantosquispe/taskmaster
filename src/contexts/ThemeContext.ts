import { createContext } from 'react';

export const Theme = {
  light: 'light',
  dark: 'dark',
  system: 'system'
} as const;

export type ThemeType = keyof typeof Theme;

export type ThemeContextType = {
  theme: ThemeType;
  setTheme: (theme: ThemeType) => void;
};

export const ThemeContext = createContext<ThemeContextType | null>(null);
