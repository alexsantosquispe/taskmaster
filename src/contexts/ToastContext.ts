import type { ToastMessage } from '@/models/types';
import { createContext } from 'react';

export interface ToastContextType {
  messages: ToastMessage[];
  addToast: (toast: ToastMessage) => void;
  removeToast: (toastId: string) => void;
}

const ToastContext = createContext<ToastContextType | null>(null);

export default ToastContext;
