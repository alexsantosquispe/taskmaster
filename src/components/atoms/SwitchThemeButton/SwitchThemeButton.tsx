import { LaptopIcon, MoonIcon, SunIcon } from '../../../icons';

import cn from 'clsx';
import { memo } from 'react';
import { twMerge } from 'tailwind-merge';
import { Theme, type ThemeType } from '../../../contexts/ThemeContext';
import { useTheme } from '../../../hooks/useTheme';
import { TabBarItem } from '../TabBar/components/TabBarItem';

const THEME_BUTTONS = [
  {
    id: Theme.system,
    label: 'System',
    icon: <LaptopIcon />
  },
  {
    id: Theme.light,
    label: 'Light',
    icon: <SunIcon />
  },
  {
    id: Theme.dark,
    label: 'Dark',
    icon: <MoonIcon />
  }
];

interface SwitchThemeButtonProps {
  isCollapsed?: boolean;
  className?: string;
}

const SwitchThemeButton = ({
  isCollapsed,
  className
}: SwitchThemeButtonProps) => {
  const { theme, setTheme } = useTheme();

  const onSelectTheme = (id: string) => {
    setTheme(id as ThemeType);
  };

  return (
    <div
      className={twMerge(
        'flex w-full self-center rounded-lg bg-neutral-100 p-0.5 text-[0.8125rem] dark:bg-neutral-800',
        cn({
          'flex-col gap-y-0.5': isCollapsed,
          'flex-row': !isCollapsed
        }),
        className
      )}
    >
      {THEME_BUTTONS.map((item) => (
        <TabBarItem
          key={item.id}
          tabItem={item}
          isSelected={theme === item.id}
          onSelectTab={() => onSelectTheme(item.id)}
          className={twMerge(
            'px-2',
            cn({
              'dark:bg-primary bg-white shadow': item.id === theme,
              'md:w-full': !isCollapsed,
              'p-1': isCollapsed
            })
          )}
          isCollapsed={isCollapsed}
        />
      ))}
    </div>
  );
};

export default memo(SwitchThemeButton);
