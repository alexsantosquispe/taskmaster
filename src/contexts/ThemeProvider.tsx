import { useState, type ReactNode } from 'react';
import { Theme, ThemeContext, type ThemeType } from './ThemeContext';

interface ThemeProviderProps {
  children: ReactNode;
}

export const ThemeProvider = ({ children }: ThemeProviderProps) => {
  const [currentTheme, setCurrentTheme] = useState<ThemeType>(Theme.system);

  return (
    <ThemeContext.Provider
      value={{
        theme: currentTheme,
        setTheme: setCurrentTheme
      }}
    >
      {children}
    </ThemeContext.Provider>
  );
};
