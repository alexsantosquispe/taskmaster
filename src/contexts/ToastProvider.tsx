import type { ToastMessage } from '@/models/types';
import { useState, type ReactNode } from 'react';
import ToastContext from './ToastContext';

interface ToastProviderProps {
  children: ReactNode;
}

export const ToastProvider = ({ children }: ToastProviderProps) => {
  const [messages, setMessages] = useState<ToastMessage[]>([]);

  const addToast = (toast: ToastMessage) => {
    setMessages((prev) => [toast, ...prev]);
  };

  const removeToast = (toastId: string) => {
    setMessages((prevMessages) =>
      prevMessages.filter((message) => message.id !== toastId)
    );
  };

  return (
    <ToastContext.Provider
      value={{
        messages,
        addToast,
        removeToast
      }}
    >
      {children}
    </ToastContext.Provider>
  );
};
