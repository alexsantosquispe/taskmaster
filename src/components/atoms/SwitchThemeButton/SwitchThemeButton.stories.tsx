import type { Meta, StoryObj } from '@storybook/react';

import { ThemeProvider } from '../../../contexts/ThemeProvider';
import SwitchThemeButton from './SwitchThemeButton';

const meta = {
  title: 'Atoms/SwitchThemeButton',
  component: SwitchThemeButton,
  parameters: {
    layout: 'centered'
  }
} satisfies Meta<typeof SwitchThemeButton>;

export default meta;

type Story = StoryObj<typeof SwitchThemeButton>;

export const Default: Story = () => {
  return (
    <ThemeProvider>
      <div className="flex flex-col gap-4">
        <SwitchThemeButton />
        <SwitchThemeButton isCollapsed={true} />
      </div>
    </ThemeProvider>
  );
};

Default.args = {};
