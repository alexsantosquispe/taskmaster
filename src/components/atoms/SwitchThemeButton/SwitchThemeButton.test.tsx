import { fireEvent, render, screen } from '@testing-library/react';

import { ThemeProvider } from '../../../contexts/ThemeProvider';
import SwitchThemeButton from './SwitchThemeButton';

describe('SwitchThemeButton', () => {
  describe('styles', () => {
    it('should render the component correctly', () => {
      let component = render(
        <ThemeProvider>
          <SwitchThemeButton />
        </ThemeProvider>
      );
      expect(component).toMatchSnapshot();

      component = render(
        <ThemeProvider>
          <SwitchThemeButton isCollapsed={true} />
        </ThemeProvider>
      );
      expect(component).toMatchSnapshot();
    });
  });

  describe('behavior', () => {
    const selectedItemStyles =
      'dark:bg-primary text-primary bg-white shadow dark:text-white';
    it('should change between tabs and check the styles are correct', () => {
      render(
        <ThemeProvider>
          <SwitchThemeButton />
        </ThemeProvider>
      );

      const systemButton = screen.getByRole('button', { name: 'System' });
      const lightButton = screen.getByRole('button', { name: 'Light' });
      const darkButton = screen.getByRole('button', { name: 'Dark' });

      expect(systemButton).toHaveClass(selectedItemStyles);
      expect(lightButton).not.toHaveClass(selectedItemStyles);
      expect(darkButton).not.toHaveClass(selectedItemStyles);

      fireEvent.click(lightButton);
      expect(lightButton).toHaveClass(selectedItemStyles);
      expect(systemButton).not.toHaveClass(selectedItemStyles);
      expect(darkButton).not.toHaveClass(selectedItemStyles);

      fireEvent.click(darkButton);
      expect(darkButton).toHaveClass(selectedItemStyles);
      expect(systemButton).not.toHaveClass(selectedItemStyles);
      expect(lightButton).not.toHaveClass(selectedItemStyles);
    });
  });
});
