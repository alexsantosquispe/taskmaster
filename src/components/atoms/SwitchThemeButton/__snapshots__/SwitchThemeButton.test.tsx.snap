// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`SwitchThemeButton styles should render the component correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex w-full self-center rounded-lg bg-neutral-100 p-0.5 text-[0.8125rem] dark:bg-neutral-800 flex-row"
      >
        <button
          aria-label="System"
          class="flex h-7 w-full items-center justify-center gap-1 rounded-md font-semibold transition-colors duration-150 ease-in-out hover:cursor-pointer text-primary dark:text-white px-2 dark:bg-primary bg-white shadow md:w-full"
        >
          <div>
            <svg
              class="size-5"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 5a2 2 0 0 1 2 2v8.526a2 2 0 0 0 .212.897l1.068 2.127a1 1 0 0 1-.9 1.45H3.62a1 1 0 0 1-.9-1.45l1.068-2.127A2 2 0 0 0 4 15.526V7a2 2 0 0 1 2-2z"
              />
              <path
                d="M20.054 15.987H3.946"
              />
            </svg>
          </div>
          <span>
            System
          </span>
        </button>
        <button
          aria-label="Light"
          class="flex h-7 w-full items-center justify-center gap-1 rounded-md font-semibold text-neutral-600 transition-colors duration-150 ease-in-out hover:cursor-pointer dark:text-white/70 px-2 md:w-full"
        >
          <div>
            <svg
              class="size-5"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="4"
              />
              <path
                d="M12 2v2"
              />
              <path
                d="M12 20v2"
              />
              <path
                d="m4.93 4.93 1.41 1.41"
              />
              <path
                d="m17.66 17.66 1.41 1.41"
              />
              <path
                d="M2 12h2"
              />
              <path
                d="M20 12h2"
              />
              <path
                d="m6.34 17.66-1.41 1.41"
              />
              <path
                d="m19.07 4.93-1.41 1.41"
              />
            </svg>
          </div>
          <span>
            Light
          </span>
        </button>
        <button
          aria-label="Dark"
          class="flex h-7 w-full items-center justify-center gap-1 rounded-md font-semibold text-neutral-600 transition-colors duration-150 ease-in-out hover:cursor-pointer dark:text-white/70 px-2 md:w-full"
        >
          <div>
            <svg
              class="size-5"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
              />
            </svg>
          </div>
          <span>
            Dark
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex w-full self-center rounded-lg bg-neutral-100 p-0.5 text-[0.8125rem] dark:bg-neutral-800 flex-row"
    >
      <button
        aria-label="System"
        class="flex h-7 w-full items-center justify-center gap-1 rounded-md font-semibold transition-colors duration-150 ease-in-out hover:cursor-pointer text-primary dark:text-white px-2 dark:bg-primary bg-white shadow md:w-full"
      >
        <div>
          <svg
            class="size-5"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 5a2 2 0 0 1 2 2v8.526a2 2 0 0 0 .212.897l1.068 2.127a1 1 0 0 1-.9 1.45H3.62a1 1 0 0 1-.9-1.45l1.068-2.127A2 2 0 0 0 4 15.526V7a2 2 0 0 1 2-2z"
            />
            <path
              d="M20.054 15.987H3.946"
            />
          </svg>
        </div>
        <span>
          System
        </span>
      </button>
      <button
        aria-label="Light"
        class="flex h-7 w-full items-center justify-center gap-1 rounded-md font-semibold text-neutral-600 transition-colors duration-150 ease-in-out hover:cursor-pointer dark:text-white/70 px-2 md:w-full"
      >
        <div>
          <svg
            class="size-5"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="4"
            />
            <path
              d="M12 2v2"
            />
            <path
              d="M12 20v2"
            />
            <path
              d="m4.93 4.93 1.41 1.41"
            />
            <path
              d="m17.66 17.66 1.41 1.41"
            />
            <path
              d="M2 12h2"
            />
            <path
              d="M20 12h2"
            />
            <path
              d="m6.34 17.66-1.41 1.41"
            />
            <path
              d="m19.07 4.93-1.41 1.41"
            />
          </svg>
        </div>
        <span>
          Light
        </span>
      </button>
      <button
        aria-label="Dark"
        class="flex h-7 w-full items-center justify-center gap-1 rounded-md font-semibold text-neutral-600 transition-colors duration-150 ease-in-out hover:cursor-pointer dark:text-white/70 px-2 md:w-full"
      >
        <div>
          <svg
            class="size-5"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
            />
          </svg>
        </div>
        <span>
          Dark
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SwitchThemeButton styles should render the component correctly 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex w-full self-center rounded-lg bg-neutral-100 p-0.5 text-[0.8125rem] dark:bg-neutral-800 flex-row"
      >
        <button
          aria-label="System"
          class="flex h-7 w-full items-center justify-center gap-1 rounded-md font-semibold transition-colors duration-150 ease-in-out hover:cursor-pointer text-primary dark:text-white px-2 dark:bg-primary bg-white shadow md:w-full"
        >
          <div>
            <svg
              class="size-5"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 5a2 2 0 0 1 2 2v8.526a2 2 0 0 0 .212.897l1.068 2.127a1 1 0 0 1-.9 1.45H3.62a1 1 0 0 1-.9-1.45l1.068-2.127A2 2 0 0 0 4 15.526V7a2 2 0 0 1 2-2z"
              />
              <path
                d="M20.054 15.987H3.946"
              />
            </svg>
          </div>
          <span>
            System
          </span>
        </button>
        <button
          aria-label="Light"
          class="flex h-7 w-full items-center justify-center gap-1 rounded-md font-semibold text-neutral-600 transition-colors duration-150 ease-in-out hover:cursor-pointer dark:text-white/70 px-2 md:w-full"
        >
          <div>
            <svg
              class="size-5"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="4"
              />
              <path
                d="M12 2v2"
              />
              <path
                d="M12 20v2"
              />
              <path
                d="m4.93 4.93 1.41 1.41"
              />
              <path
                d="m17.66 17.66 1.41 1.41"
              />
              <path
                d="M2 12h2"
              />
              <path
                d="M20 12h2"
              />
              <path
                d="m6.34 17.66-1.41 1.41"
              />
              <path
                d="m19.07 4.93-1.41 1.41"
              />
            </svg>
          </div>
          <span>
            Light
          </span>
        </button>
        <button
          aria-label="Dark"
          class="flex h-7 w-full items-center justify-center gap-1 rounded-md font-semibold text-neutral-600 transition-colors duration-150 ease-in-out hover:cursor-pointer dark:text-white/70 px-2 md:w-full"
        >
          <div>
            <svg
              class="size-5"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
              />
            </svg>
          </div>
          <span>
            Dark
          </span>
        </button>
      </div>
    </div>
    <div>
      <div
        class="flex w-full self-center rounded-lg bg-neutral-100 p-0.5 text-[0.8125rem] dark:bg-neutral-800 flex-col gap-y-0.5"
      >
        <button
          aria-label="System"
          class="flex h-7 w-full items-center justify-center gap-1 rounded-md font-semibold transition-colors duration-150 ease-in-out hover:cursor-pointer md:w-24 text-primary dark:text-white dark:bg-primary bg-white shadow p-1"
        >
          <div>
            <svg
              class="size-5"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 5a2 2 0 0 1 2 2v8.526a2 2 0 0 0 .212.897l1.068 2.127a1 1 0 0 1-.9 1.45H3.62a1 1 0 0 1-.9-1.45l1.068-2.127A2 2 0 0 0 4 15.526V7a2 2 0 0 1 2-2z"
              />
              <path
                d="M20.054 15.987H3.946"
              />
            </svg>
          </div>
        </button>
        <button
          aria-label="Light"
          class="flex h-7 w-full items-center justify-center gap-1 rounded-md font-semibold text-neutral-600 transition-colors duration-150 ease-in-out hover:cursor-pointer md:w-24 dark:text-white/70 p-1"
        >
          <div>
            <svg
              class="size-5"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="4"
              />
              <path
                d="M12 2v2"
              />
              <path
                d="M12 20v2"
              />
              <path
                d="m4.93 4.93 1.41 1.41"
              />
              <path
                d="m17.66 17.66 1.41 1.41"
              />
              <path
                d="M2 12h2"
              />
              <path
                d="M20 12h2"
              />
              <path
                d="m6.34 17.66-1.41 1.41"
              />
              <path
                d="m19.07 4.93-1.41 1.41"
              />
            </svg>
          </div>
        </button>
        <button
          aria-label="Dark"
          class="flex h-7 w-full items-center justify-center gap-1 rounded-md font-semibold text-neutral-600 transition-colors duration-150 ease-in-out hover:cursor-pointer md:w-24 dark:text-white/70 p-1"
        >
          <div>
            <svg
              class="size-5"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex w-full self-center rounded-lg bg-neutral-100 p-0.5 text-[0.8125rem] dark:bg-neutral-800 flex-col gap-y-0.5"
    >
      <button
        aria-label="System"
        class="flex h-7 w-full items-center justify-center gap-1 rounded-md font-semibold transition-colors duration-150 ease-in-out hover:cursor-pointer md:w-24 text-primary dark:text-white dark:bg-primary bg-white shadow p-1"
      >
        <div>
          <svg
            class="size-5"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 5a2 2 0 0 1 2 2v8.526a2 2 0 0 0 .212.897l1.068 2.127a1 1 0 0 1-.9 1.45H3.62a1 1 0 0 1-.9-1.45l1.068-2.127A2 2 0 0 0 4 15.526V7a2 2 0 0 1 2-2z"
            />
            <path
              d="M20.054 15.987H3.946"
            />
          </svg>
        </div>
      </button>
      <button
        aria-label="Light"
        class="flex h-7 w-full items-center justify-center gap-1 rounded-md font-semibold text-neutral-600 transition-colors duration-150 ease-in-out hover:cursor-pointer md:w-24 dark:text-white/70 p-1"
      >
        <div>
          <svg
            class="size-5"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="4"
            />
            <path
              d="M12 2v2"
            />
            <path
              d="M12 20v2"
            />
            <path
              d="m4.93 4.93 1.41 1.41"
            />
            <path
              d="m17.66 17.66 1.41 1.41"
            />
            <path
              d="M2 12h2"
            />
            <path
              d="M20 12h2"
            />
            <path
              d="m6.34 17.66-1.41 1.41"
            />
            <path
              d="m19.07 4.93-1.41 1.41"
            />
          </svg>
        </div>
      </button>
      <button
        aria-label="Dark"
        class="flex h-7 w-full items-center justify-center gap-1 rounded-md font-semibold text-neutral-600 transition-colors duration-150 ease-in-out hover:cursor-pointer md:w-24 dark:text-white/70 p-1"
      >
        <div>
          <svg
            class="size-5"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
            />
          </svg>
        </div>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
